Lib/dataclasses.py

```diff
@@ -497,12 +519,12 @@ def _init_fn(fields, frozen, has_post_init, self_name):
 
 
 def _repr_fn(fields):
-    return _create_fn('__repr__',
-                      ('self',),
-                      ['return self.__class__.__qualname__ + f"(' +
-                       ', '.join([f"{f.name}={{self.{f.name}!r}}"
-                                  for f in fields]) +
-                       ')"'])
+    return _recursive_repr(_create_fn('__repr__',
+                                      ('self',),
+                                      ['return self.__class__.__qualname__ + f"(' +
+                                       ', '.join([f"{f.name}={{self.{f.name}!r}}"
```

Comment:
I suspect these lines are too long, but I haven't looked in a real editor yet. I'm not sure if it's worth the hassle to reflow this, or not. Maybe just introduce a temporary variable to hold the return value of `_create_fn`?
```
    repr = _create_fn(...
                     )
    return _recursive_repr(repr)
```